diff --git a/PKGBUILD b/PKGBUILD
index 3a7feed..507a156 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -21,7 +21,7 @@ makedepends=('git' 'bison' 'flex' 'asciidoc' 'xmlto' 'meson')
 optdepends=('i3lock: For locking your screen.'
             'i3status: To display system information with a bar.')
 options=('docs')
-source=('git+https://github.com/i3/i3#branch=next')
+source=('git+https://github.com/i3/i3#branch=next' 'rounded-corners.patch')
 sha1sums=('SKIP')
 
 pkgver() {
@@ -46,4 +46,11 @@ package() {
   install -Dt "${pkgdir}/usr/share/licenses/${pkgname}" -m644 LICENSE
 }
 
+prepare() {
+  cd "i3/src"
+  patch --forward --input=../../rounded-corners.patch
+}
+
 # vim:set ts=2 sw=2 et:
+sha1sums=('SKIP'
+          '58a37b0d78fa0814cb289329d743923203be9d18')
