diff --git a/PKGBUILD b/PKGBUILD
index 24fdffb..0d45254 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -11,8 +11,8 @@ conflicts=('svkbd')
 provides=('svkbd')
 makedepends=('git' 'discount')
 license=('MIT')
-source=('git://git.suckless.org/svkbd')
-sha256sums=('SKIP')
+source=('git://git.suckless.org/svkbd' 'layout.hr.h')
+sha256sums=('SKIP' 'SKIP')
 
 pkgver() {
 	cd svkbd/
@@ -29,10 +29,14 @@ pkgver() {
 	fi
 }
 
+prepare() {
+	mv ../layout.hr.h svkbd/
+}
+
 build() {
 	cd svkbd/
 
-	make
+	make LAYOUT=hr
 
 	markdown README.md > README.html
 }
@@ -40,7 +44,7 @@ build() {
 package() {
 	cd svkbd/
 
-	make DESTDIR="${pkgdir}" PREFIX='/usr/' install
+	make LAYOUT=hr DESTDIR="${pkgdir}" PREFIX='/usr/' install
 
 	install -D -m0644 'LICENSE' "${pkgdir}/usr/share/licenses/svkbd/LICENSE"
 	install -D -m0644 'README.md' "${pkgdir}/usr/share/doc/svkbd/README.md"
