diff --git a/source/view.c b/source/view.c
index 995a79ef..ff0d1fac 100644
--- a/source/view.c
+++ b/source/view.c
@@ -639,17 +639,7 @@ void rofi_view_set_active(RofiViewState *state) {
 void rofi_view_set_selected_line(RofiViewState *state,
                                  unsigned int selected_line) {
   state->selected_line = selected_line;
-  // Find the line.
-  unsigned int selected = 0;
-  for (unsigned int i = 0; ((state->selected_line)) < UINT32_MAX && !selected &&
-                           i < state->filtered_lines;
-       i++) {
-    if (state->line_map[i] == (state->selected_line)) {
-      selected = i;
-      break;
-    }
-  }
-  listview_set_selected(state->list_view, selected);
+  listview_set_selected(state->list_view, selected_line);
   xcb_clear_area(xcb->connection, CacheState.main_window, 1, 0, 0, 1, 1);
   xcb_flush(xcb->connection);
 }
diff --git a/source/widgets/listview.c b/source/widgets/listview.c
index b76bedee..663234f7 100644
--- a/source/widgets/listview.c
+++ b/source/widgets/listview.c
@@ -647,7 +647,7 @@ void listview_set_selected(listview *lv, unsigned int selected) {
     return;
   }
   if (lv->req_elements > 0) {
-    lv->selected = MIN(selected, lv->req_elements - 1);
+    lv->selected = selected;
     lv->barview.direction = LEFT_TO_RIGHT;
     widget_queue_redraw(WIDGET(lv));
     if (lv->sc_callback) {
