diff --git a/PKGBUILD b/PKGBUILD
index dec78d3..f0c5008 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -63,6 +63,10 @@ prepare() {
 	git config submodule.subprojects/libgwater.url "${srcdir}/libgwater"
 	git config submodule.subprojects/libnkutils.url "${srcdir}/libnkutils"
 	git -c protocol.file.allow=always submodule update
+
+	cd "${srcdir}/rofi"
+	git apply ../../set-selected-fix.patch
+
 	cd "${srcdir}"
 }
 
