diff --git a/backlight_control.c b/backlight_control.c
index 2a04932..d0630e0 100644
--- a/backlight_control.c
+++ b/backlight_control.c
@@ -14,7 +14,7 @@
 
 void print_usage(char *name) {
 	printf(
-		"Usage: %1$s [+|-]<value>\n"
+		"Usage: %1$s [[+|-]<value>]\n"
 		"\n"
 		"Examples:\n"
 		"\t%1$s +10\n"
@@ -34,13 +34,20 @@ FILE *open_file(char *name) {
 }
 
 int main(int argc, char **argv) {
+  	FILE *brightness = open_file(BACKLIGHT_DIR "brightness");
+	int brightness_value = MIN_BRIGHTNESS;
+	if (argc == 1) {
+		fscanf(brightness, "%d", &brightness_value);
+		brightness_value = (int)((double)brightness_value / MAX_BRIGHTNESS * 100);
+		fprintf(stdout, "%d\n", brightness_value);
+		fclose(brightness);
+		return EXIT_SUCCESS;
+	}
 	if (argc != 2) {
 		print_usage(argv[0]);
 		return EXIT_FAILURE;
 	}
 	int value = strtol(argv[1], NULL, 10);
-	FILE *brightness = open_file(BACKLIGHT_DIR "brightness");
-	int brightness_value = MIN_BRIGHTNESS;
 	switch (argv[1][0]) {
 		case '+':
 		case '-':
@@ -53,6 +60,8 @@ int main(int argc, char **argv) {
 	brightness_value = MIN(brightness_value, MAX_BRIGHTNESS);
 	brightness_value = MAX(brightness_value, MIN_BRIGHTNESS);
 	fprintf(brightness, "%d", brightness_value);
+	brightness_value = (int)((double)brightness_value / MAX_BRIGHTNESS * 100);
+	fprintf(stdout, "%d\n", brightness_value);
 	fclose(brightness);
 	return EXIT_SUCCESS;
 }
